packages/theme-product-form/__tests__/theme-product-form.test.js

```diff
@@ -0,0 +1,311 @@
+    };
+    const quantity = 10;
+
+    // eslint-disable-next-line no-unused-vars
+    const productForm = new ProductForm(element, productJSON, options);
+
+    quantityElement.value = quantity;
+    quantityElement.dispatchEvent(changeEvent);
+
+    expect(options.onQuantityChange).toHaveBeenCalledWith(changeEvent);
+    expectFormEventDataset(changeEvent, {quantity});
+  });
+
+  test('calls the method assigned to onPropertyChange option when the value of a property input changes', () => {
+    const element = document.getElementById('form');
+    const propertyElement = document.querySelector('[name^="properties"]');
+    const changeEvent = new Event('change');
+    const options = {
+      onPropertyChange: jest.fn()
+    };
+    const properties = [
+      {name: 'Message', value: 'doh'},
+      {name: 'Hidden', value: 'something'},
+      {name: 'Subscribe', value: 'true'}
+    ];
+
+    // eslint-disable-next-line no-unused-vars
+    const productForm = new ProductForm(element, productJSON, options);
+
+    propertyElement.value = properties[0].value;
+    propertyElement.dispatchEvent(changeEvent);
+
+    expect(options.onPropertyChange).toHaveBeenCalledWith(changeEvent);
+    expectFormEventDataset(changeEvent, {properties});
+  });
+
+  test('calls the method assigned to onFormSubmit option when the form is submitted', () => {
+    const element = document.getElementById('form');
+    const submitEvent = new Event('submit');
+    const options = {
+      onFormSubmit: jest.fn()
+    };
+
+    const productForm = new ProductForm(element, productJSON, options);
+
+    productForm.element.dispatchEvent(submitEvent);
+
+    expect(options.onFormSubmit).toHaveBeenCalledWith(submitEvent);
+    expectFormEventDataset(submitEvent, {});
+  });
+});
+
+describe('ProductForm.destroy()', () => {
+  test('removes all event listeners that were assigned by ProductForm()', () => {
+    const element = document.getElementById('form');
+    const colorSelect = document.querySelector('[name="options[Color]"]');
+    const quantityElement = document.querySelector('[name="quantity"]');
+    const propertyElement = document.querySelector('[name^="properties"]');
+
+    const changeEvent = new Event('change');
+    const submitEvent = new Event('submit');
+
+    const config = {
+      onOptionChange: jest.fn(),
+      onQuantityChange: jest.fn(),
+      onPropertyChange: jest.fn(),
+      onFormSubmit: jest.fn()
+    };
+
+    const productForm = new ProductForm(element, productJSON, config);
+
+    productForm.destroy();
+
+    colorSelect.dispatchEvent(changeEvent);
+    quantityElement.dispatchEvent(changeEvent);
+    propertyElement.dispatchEvent(changeEvent);
+    productForm.element.dispatchEvent(submitEvent);
+
+    expect(config.onOptionChange).not.toHaveBeenCalled();
+    expect(config.onQuantityChange).not.toHaveBeenCalled();
+    expect(config.onPropertyChange).not.toHaveBeenCalled();
+    expect(config.onFormSubmit).not.toHaveBeenCalled();
```

Comment:
I know this is using `not`, but why do these not need `changeEvent` or `submitEvent` like these ones do?
`expect(options.onFormSubmit).toHaveBeenCalledWith(submitEvent)`

Reply:
We're testing destroy(), so I want to make sure these functions where not called.